Python/Product/Analysis/ModuleResolver.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PythonTools.Analysis.Infrastructure;
+using Microsoft.PythonTools.Parsing.Ast;
+
+namespace Microsoft.PythonTools.Analysis {
+    internal class ModuleResolver {
+        /// <summary>
+        /// Resolves modules listed in the relative import statetement
+        /// such as 'from . import a, b, c' or 'from .a import b, c'
+        /// </summary>
+        /// <param name="entry">Project entry</param>
+        /// <param name="node">Import statement node</param>
+        /// <returns>Names of modules</returns>
+        internal static IEnumerable<string> ResolveRelativeFromImport(IPythonProjectEntry entry, FromImportStatement node)
+            => ResolveRelativeFromImport(entry.ModuleName, entry.FilePath, node);
+
+        /// <summary>
+        /// Resolves modules listed in the relative import statetement
+        /// such as 'from . import a, b, c' or 'from .a import b, c'
+        /// </summary>
+        /// <param name="importingFromModuleName">Name of the importing module</param>
+        /// <param name="importingFromFilePath">Disk path to the importing module</param>
+        /// <param name="node">Import statement node</param>
+        /// <returns></returns>
+        internal static IEnumerable<string> ResolveRelativeFromImport(string importingFromModuleName, string importingFromFilePath, FromImportStatement node) {
+            var root = node.Root.MakeString();
+
+            if (!string.IsNullOrEmpty(root) && root.StartsWith(".")) {
+                var prefix = root.All(c => c == '.') ? root : $"{root}.";
```

Comment:
`$` (culture-sensitive formatting) should not be used in this context - use `"{0}.".FormatInvariant(root)` instead